api/envoy/config/core/v3/substitution_format_string.proto

```diff
@@ -58,4 +58,11 @@ message SubstitutionFormatString {
     //
     google.protobuf.Struct json_format = 2 [(validate.rules).message = {required: true}];
   }
+
+  // If set to true, when command operators are evaluated to null,
+  //
+  // * for ``text_format``, the output of the empty operator is changed from ``-`` to an
+  //   empty string, so that empty values are omitted entirely.
+  // * for ``json_format`` the keys with null values are omitted in the output structure.
+  bool omit_empty_values = 3;
```

Comment:
Seems reasonable. Do you think there will be use cases where there is a need for an empty marker other than `""` or `"-"`?

Reply:
My main goal was to omit the null values from the JSON format together with the keys while changing the text_format is more of a side effect, to make the API more consistent within itself. So, I didn't think over the text_format extensively, but I feel like `-` and empty string should be enough. 

A few plaintext formats from the top of my head:
- CSV would benefit from empty string, as null is represented as `,,` - thus the empty string would be useful.
- syslog NILVALUE is `-`